lib/runtime.js

```diff
@@ -9,6 +9,7 @@
  * @typedef {import('./core.js').core} Core
  *
  * @typedef {Record<string, HPropertyValue | HStyle | HChild>} JSXProps
+ * @typedef {(props: JSXProps) => HResult } HFn
```

Comment:
```suggestion
 * @typedef {(props: JSXProps) => HResult} HFn
```

I think I prefer a name such as `HFunctionComponent` more?